node_modules/col-assets/lib/api.js

```diff
@@ -957,75 +958,135 @@ var canDeleteAsset = function(user, asset) {
     assetUrl = asset.url;
   }
 
-  if (!config.get('embdr.enabled') || !assetUrl || asset.embed_id) {
+  if (!config.get('previews.enabled') || !assetUrl || asset.embed_id) {
     return false;
   }
 
-  var embdr = new Embdr(config.get('embdr.apiKey'));
-  embdr.process(assetUrl, {
-    'start': function(preview) {
-      updateAssetPreview(asset, {
-        'embedId': preview.id,
-        'embedKey': preview.embedKey,
-        'embedCode': preview.embedCode
-      });
-    },
-    'thumbnails': {
-      'sizes': [CollabosphereConstants.THUMBNAIL_SIZE],
-      'complete': function(thumbnails) {
-        var thumbnail = thumbnails[CollabosphereConstants.THUMBNAIL_SIZE];
-        if (thumbnail && thumbnail.url) {
-          updateAssetPreview(asset, {'thumbnailUrl': thumbnail.url});
-        }
-      }
+  var protocol = config.get('app.https') ? 'https' : 'http';
+  var authorizationHeader = Collabosphere.generatePreviewServiceSignature();
+  var params = {
+    'form': {
+      'id': asset.id,
+      'url': assetUrl,
+      'postBackUrl': util.format('%s://%s/api/assets-callback', protocol, config.get('app.host'))
     },
-    'images': {
-      'sizes': [CollabosphereConstants.IMAGE_SIZE],
-      'complete': function(images) {
-        // Update all whiteboard elements that use this asset
-        var image = images[CollabosphereConstants.IMAGE_SIZE];
-        if (image && image.url) {
-          require('col-whiteboards').updateAssetPreview(asset.id, image);
-          updateAssetPreview(asset, {'imageUrl': image.url});
-        }
-      }
-    },
-    'error': function(err) {
-      return log.error({'err': err, 'asset': asset.id}, 'An error occurred when generating an asset preview');
+    'headers': {
+      'authorization': authorizationHeader
+    }
+  };
+  request.post(config.get('previews.url'), params, function(err, response, body) {
+    if (err) {
+      log.error({'err': err, 'asset': asset.id}, 'An error occurred when generating previews');
+      updateAssetPreview(asset, {'previewStatus': 'error'});
+      return callback({'code': 500, 'msg': 'Failed to contact the previews service'});
+    } else if (response.statusCode !== 200) {
+      log.error({'err': err, 'asset': asset.id}, 'An error occurred when generating previews');
+      updateAssetPreview(asset, {'previewStatus': 'error'});
+      return callback({'code': 500, 'msg': 'The previews service returned an error'});
     }
+
+    return callback();
   });
 };
 
 /**
- * Update the preview metadata for an asset
+ * Handle a callback from the previews service
+ *
+ * @param  {String}       authorizationHeader     The authorization header
+ * @param  {Object}       opts                    The passed in data from the previews service
+ * @param  {Number}       opts.id                 The asset id
+ * @param  {String}       opts.status             The status of the preview process
+ * @param  {String}       opts.metadata           Extra preview metadata
```

Comment:
It's only used for link processing. Specifically:

 - youtubeId - The youtube ID in case it's a Youtube video
 - httpEmbeddable - Whether the link can be embedded in an iframe over http
 - httpsEmbeddable - Whether the link can be embedded in an iframe over https